# **Customer Churn**

<b><center>Analysis on: 31th May 2024 <br> Gagandev Mishra </center></b>

When the customer stops using their products or services due to any reason, then it's called customer churned, where the business has lost a valuable or prospective customer who can be profitable for the establishment, but due to any reason now the customer is not engaged with the business, and that is not what the business wants.

So, here you will see what the normal customer looks like in terms of demographics, the type of service they have enrolled in, what made the customer  end their service with the business, and many more important aspects you will find once you run down through the steps.

But before that, let me brief you on the dataset on which I'm going to analyze it.

## Dataset Background

This dataset belongs to one of the fictitious telecom providers that provide telecom services to their California-based customers, which I downloaded from Github.
Originally,  it belonged to IBM, but no name was mentioned for the telecom provider, so I put Fictel as the provider name for reference in my analysis.

After downloading the dataset, I converted the file into a CSV and added it to the PostgreSQL database so that I could analyze it with the SQL query.

The table in which I'm going to work as follows:
1. Telco_customer_churn_services
2. Telco_customer_churn_status
3. Telco_customer_churn_population
4. Telco_customer_churn_location
5. Telco_customer_churn_demographics

To learn more about data you can refer to this [data dictionary](https://community.ibm.com/community/user/businessanalytics/blogs/steven-macko/2019/07/11/telco-customer-churn-1113)

## Connect Data

First, we will install the necessary libraries so that we can connect with the PostgreSQL database on the Jupyter notebook, and then we will continue ahead.

In [ ]:
!pip install ipython-sql
!pip install sqlalchemy
!pip install psycopg2

Now load the SQL extension.

In [ ]:
%load_ext sql

This is the last command to execute to connect with PostgreSQL, and after that, you just need to type '%%sql' before putting your SQL query.

In [ ]:
%sql postgresql://postgres:**********@localhost:5432/telco

## Data Oveview

Whenever we start with analysis, the first thing we should do is get familiar with the data, and here also we will try to get familiar with the datasets.

To get acquitance with the data, here is the SQL query, which will show you all the tables with their names, the number of rows and columns they have, along with the available column names from the respective table, so that you know which table has the required data.

In [ ]:
%%sql

WITH data_overview AS (
SELECT 'Churn Status' AS table_name, COUNT(*) AS total_row,
		(SELECT COUNT(*)
		   FROM INFORMATION_SCHEMA.COLUMNS
		  WHERE table_name = 'customer_churn_status') AS total_columns,
		(SELECT STRING_AGG(column_name, ', ')
		   FROM INFORMATION_SCHEMA.COLUMNS
		  WHERE table_name = 'customer_churn_status') AS column_name
  FROM customer_churn_status
 UNION ALL
SELECT 'Churn Location' AS table_name, COUNT(*)AS total_row,
		(SELECT COUNT(*)
		   FROM INFORMATION_SCHEMA.COLUMNS
		  WHERE table_name = 'customer_churn_location') AS total_columns,
		(SELECT STRING_AGG(column_name, ', ')
		   FROM INFORMATION_SCHEMA.COLUMNS
		  WHERE table_name = 'customer_churn_location') AS column_name
  FROM customer_churn_location
 UNION ALL
SELECT 'Churn Population' AS table_name, COUNT(*)AS total_row,
		(SELECT COUNT(*)
		   FROM INFORMATION_SCHEMA.COLUMNS
		  WHERE table_name = 'customer_churn_population') AS total_columns,
		(SELECT STRING_AGG(column_name, ', ')
		   FROM INFORMATION_SCHEMA.COLUMNS
		  WHERE table_name = 'customer_churn_population') AS column_name
  FROM customer_churn_population
 UNION ALL
SELECT 'Churn Service' AS table_name, COUNT(*)AS total_row,
		(SELECT COUNT(*)
		   FROM INFORMATION_SCHEMA.COLUMNS
		  WHERE table_name = 'customer_churn_service') AS total_columns,
		(SELECT STRING_AGG(column_name, ', ')
		   FROM INFORMATION_SCHEMA.COLUMNS
		  WHERE table_name = 'customer_churn_service') AS column_name
  FROM customer_churn_service
 UNION ALL
SELECT 'Churn Demographic' AS table_name, COUNT(*)AS total_row,
		(SELECT COUNT(*)
		   FROM INFORMATION_SCHEMA.COLUMNS
		  WHERE table_name = 'customer_churn_demographic') AS total_columns,
		(SELECT STRING_AGG(column_name, ', ')
		   FROM INFORMATION_SCHEMA.COLUMNS
		  WHERE table_name = 'customer_churn_demographic') AS column_name
  FROM customer_churn_status

)

SELECT *
  FROM data_overview
 ORDER BY total_columns ASC;

 * postgresql://postgres:***@localhost:5432/telco
5 rows affected.


[('Churn Population', 1671, 3, 'id, zip_code, population'),
 ('Churn Location', 7043, 9, 'zipcode, customer_count, latitude, longitude, lat_long, city, country, states, customer_id'),
 ('Churn Demographic', 7043, 9, 'customer_count, age, dependents_number, senior_citizen, married, dependents, under_30, gender, customer_id'),
 ('Churn Status', 7043, 11, 'churn_value, churn_score, cltv, customer_count, satisfication_score, customer_id, churn_reason, quarter, customer_status, churn_label, churn_category'),
 ('Churn Service', 7043, 31, 'tenure_in_month, customer_count, no_referred, avg_monthly_gb_download, avg_monthly_long_distance, monthly_charges, total_charges, total_refund, total ... (223 characters truncated) ... rred_friend, offer, phone_service, multiple_lines, internet_service, internet_type, online_security, online_backup, customer_id, premium_tech_support')]

As I have initially stated, there are 6 different tables, each carrying 7043 rows, except one table has just 1671 rows of data.

After getting the size of the data, we are ready to move on to the next step, where we will check whether any row or column has a duplicate and null value.

### Check Duplicate Value

There are various data points present in the rows, and from that, I'll try to find out if any duplicate values exist in the data.
While exploring the data, I found the `customer_id` column, which should not have any duplicate values in it.
If it is present, then I have to find some solution to fix it.

Now let type the query for it.

In [ ]:
%%sql
 SELECT csr.customer_id AS churn_service_customer_id,
	     ccs.customer_id AS churn_status_customer_id,
	     ccd.customer_id AS churn_demographic_customer_id,
	     ccl.customer_id AS churn_location_customer_id
   FROM customer_churn_service AS csr
  INNER JOIN customer_churn_status AS ccs
     ON csr.customer_id = ccs.customer_id
  INNER JOIN customer_churn_demographic AS ccd
     ON csr.customer_id = ccd.customer_id
  INNER JOIN customer_churn_location AS ccl
     ON csr.customer_id = ccl.customer_id
  GROUP BY churn_service_customer_id, churn_status_customer_id, churn_demographic_customer_id, churn_location_customer_id
 HAVING COUNT(*) > 1;

 * postgresql://postgres:***@localhost:5432/telco
0 rows affected.


[]

As the result shows, there is no duplicate value in the data, which is a good thing to see.

### Any null value exist in any columns


While checking for the null value, I have not found any null values in `churn_population`, `churn_location`, `churn_demographic`, and `churn_service`, but there is a null value in the `churn_status` table in a `churn_category` and a `churn_reason` columns.

In [ ]:
%%sql
SELECT *
  FROM customer_churn_status
 WHERE customer_id IS NULL
    OR customer_count IS NULL
	OR quarter IS NULL
	OR satisfication_score IS NULL
	OR customer_status IS NULL
	OR churn_label IS NULL
	OR churn_value IS NULL
	OR churn_score IS NULL
	OR cltv IS NULL
	OR churn_category IS NULL
	OR churn_reason IS NULL
LIMIT 10;

 * postgresql://postgres:***@localhost:5432/telco
10 rows affected.


[('7251-LJBQN', 1, 'Q3', 3, 'Stayed', 'No', 0, 20, 3568, None, None),
 ('2682-KEVRP', 1, 'Q3', 3, 'Stayed', 'No', 0, 21, 4327, None, None),
 ('9978-HYCIN', 1, 'Q3', 3, 'Stayed', 'No', 0, 68, 3384, None, None),
 ('8329-UTMVM', 1, 'Q3', 4, 'Stayed', 'No', 0, 31, 3104, None, None),
 ('6952-OMNWB', 1, 'Q3', 4, 'Stayed', 'No', 0, 28, 5581, None, None),
 ('6549-YMFAW', 1, 'Q3', 4, 'Stayed', 'No', 0, 69, 2302, None, None),
 ('1307-TVUFB', 1, 'Q3', 4, 'Stayed', 'No', 0, 79, 2399, None, None),
 ('7503-MIOGA', 1, 'Q3', 3, 'Stayed', 'No', 0, 56, 5881, None, None),
 ('2499-AJYUA', 1, 'Q3', 5, 'Stayed', 'No', 0, 80, 5933, None, None),
 ('6919-ELBGL', 1, 'Q3', 3, 'Stayed', 'No', 0, 61, 6268, None, None)]

After looking at the at the output, it made me aware of the existence of a null value because when the customer is not churned, it's not possible to have their response in `churn_category` and `churn_label`.

In [ ]:
%%sql
SELECT COUNT(*)
  FROM customer_churn_status
 WHERE churn_category IS NULL
   AND churn_reason IS NULL
   AND churn_label = 'No';

 * postgresql://postgres:***@localhost:5432/telco
1 rows affected.


[(5174,)]

#### Handling Null Values

There is a total of 5174 where the data is missing, which we can keep as it is, but I'll update the null value with NA so that it is easier for me to recognize.

I have two options: I can use the `COALESCE()` function to fill a null value with NA, and in the other option, I can replace the null value directly in the table with NA.

But I'm going with the second option, which I find suitable for this scenario.

In [ ]:
%%sql
--# Replace null value with NA in churn_reason, churn_category
UPDATE customer_churn_status
   SET churn_category = 'NA', churn_reason = 'NA'
 WHERE churn_category IS NULL AND churn_reason IS NULL AND churn_label = 'No';

 * postgresql://postgres:***@localhost:5432/telco
0 rows affected.


[]

Let's verify the changes.

In [ ]:
%%sql
SELECT COUNT(*)
  FROM customer_churn_status
 WHERE churn_category IS NULL
   AND churn_reason IS NULL
   AND churn_label = 'No';

 * postgresql://postgres:***@localhost:5432/telco
1 rows affected.


[(0,)]

At this point, there are no more null values in the data.

### Data Range

The next thing I will do is check the data range so that I can know what attributes are available, which will eventually help me find any anomalies in the data.

In [ ]:
%%sql
SELECT STRING_AGG( DISTINCT gender, ', ') AS gender_range,
		CONCAT(MIN(age), '-' ,MAX(age)) AS age_range,
		PERCENTILE_CONT(0.50) WITHIN GROUP (ORDER BY age) AS median_age,
 		STRING_AGG(DISTINCT under_30, ', ') AS under_30_range,
 		STRING_AGG(DISTINCT senior_citizen, ', ') AS senior_citize_range,
		STRING_AGG(DISTINCT married, ', ') AS married_range,
		STRING_AGG(DISTINCT CAST(dependents AS TEXT), ', ') AS depndents_range
  FROM customer_churn_demographic;

 * postgresql://postgres:***@localhost:5432/telco
1 rows affected.


[('Female, Male', '19-80', 46.0, 'No, Yes', 'No, Yes', 'No, Yes', 'No, Yes')]

Using the above query, I can check the range of data in every table, but I want to stack the information from other tables one by one, which will help me get a picture of the data by just scrolling down in the query results.

In [ ]:
%%sql
SELECT 'gender_range' AS "Column Name",
        STRING_AGG(DISTINCT gender, ', ') AS value
  FROM customer_churn_demographic
 UNION ALL
SELECT 'age_range',
        CONCAT(MIN(age), '-', MAX(age))
  FROM customer_churn_demographic
 UNION ALL
SELECT 'under_30_range',
        STRING_AGG(DISTINCT under_30, ', ')
  FROM customer_churn_demographic
 UNION ALL
SELECT 'senior_citize_range',
        STRING_AGG(DISTINCT senior_citizen, ', ')
  FROM customer_churn_demographic
 UNION ALL
SELECT 'married_range',
        STRING_AGG(DISTINCT married, ', ')
  FROM customer_churn_demographic
 UNION ALL
SELECT 'dependents_range',
        STRING_AGG(DISTINCT CAST(dependents AS TEXT), ', ')
  FROM customer_churn_demographic
 UNION ALL
SELECT 'country',
        STRING_AGG(DISTINCT country, ', ')
  FROM customer_churn_location
 UNION ALL
SELECT 'state',
        STRING_AGG(DISTINCT states, ', ')
  FROM customer_churn_location
 UNION ALL
SELECT 'zip_code_range',
        CONCAT(MIN(zipcode), ' - ', MAX(zipcode))
  FROM customer_churn_location
 UNION ALL
SELECT 'lat_long_range',
        CONCAT(MIN(lat_long), ' - ', MAX(lat_long))
  FROM customer_churn_location
 UNION ALL
SELECT 'quarter',
		   STRING_AGG(DISTINCT quarter, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'referred_friend',
		   STRING_AGG(DISTINCT referred_friend, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'no_referred',
		   CONCAT(MIN(no_referred), ' - ', MAX(no_referred))
  FROM customer_churn_service
 UNION ALL
SELECT 'tenure_in_month',
		   CONCAT(MIN(tenure_in_month), ' - ', MAX(tenure_in_month))
  FROM customer_churn_service
 UNION ALL
SELECT 'offer',
		   STRING_AGG(DISTINCT offer, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'phone_service',
		   STRING_AGG(DISTINCT phone_service, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'avg_monthly_long_distance',
		   CONCAT(MIN(avg_monthly_long_distance), ' - ', MAX(avg_monthly_long_distance))
  FROM customer_churn_service
 UNION ALL
SELECT 'multiple_lines',
		   STRING_AGG(DISTINCT multiple_lines, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'internet_service',
		    STRING_AGG(DISTINCT internet_service, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'internet_type',
		    STRING_AGG(DISTINCT internet_type, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'avg_monthly_gb_download',
		   CONCAT(MIN(avg_monthly_gb_download), ' - ', MAX(avg_monthly_gb_download))
  FROM customer_churn_service
 UNION ALL
SELECT 'online_security',
		   STRING_AGG(DISTINCT online_security, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'online_backup',
		    STRING_AGG(DISTINCT online_backup, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'device_protection',
		    STRING_AGG(DISTINCT device_protection, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'premium_tech_support',
		    STRING_AGG(DISTINCT premium_tech_support, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'streaming_tv',
		    STRING_AGG(DISTINCT streaming_tv, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'streaming_movies',
		    STRING_AGG(DISTINCT streaming_movies, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'streaming_music',
		    STRING_AGG(DISTINCT streaming_music, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'unlimited_data',
		    STRING_AGG(DISTINCT unlimited_data, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'contract',
		    STRING_AGG(DISTINCT contract, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'paperless_billing',
		    STRING_AGG(DISTINCT paperless_billing, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'payment_method',
		    STRING_AGG(DISTINCT payment_method, ', ')
  FROM customer_churn_service
 UNION ALL
SELECT 'monthly_charges',
		    CONCAT(MIN(monthly_charges), ' - ', MAX(monthly_charges))
  FROM customer_churn_service
 UNION ALL
SELECT 'total_charges',
		    CONCAT(MIN(total_charges), ' - ', MAX(total_charges))
  FROM customer_churn_service
 UNION ALL
SELECT 'total_refund',
		    CONCAT(MIN(total_refund), ' - ', MAX(total_refund))
  FROM customer_churn_service
 UNION ALL
SELECT 'total_extra_data_charges',
		    CONCAT(MIN(total_extra_data_charges), ' - ', MAX(total_extra_data_charges))
  FROM customer_churn_service
 UNION ALL
SELECT 'total_long_distance_charges',
		   CONCAT(MIN(total_long_distance_charges), ' - ', MAX(total_long_distance_charges))
  FROM customer_churn_service
 UNION ALL
SELECT 'total_revenue',
		   CONCAT(MIN(total_revenue), ' - ', MAX(total_revenue))
  FROM customer_churn_service
 UNION ALL
SELECT 'satisfication_score', CONCAT(MIN(satisfication_score), ' - ' , MAX(satisfication_score))
  FROM customer_churn_status
 UNION ALL
SELECT 'customer_status', STRING_AGG(DISTINCT customer_status, ', ')
  FROM customer_churn_status
 UNION ALL
SELECT 'churn_label', STRING_AGG(DISTINCT churn_label, ', ')
  FROM customer_churn_status
 UNION ALL
SELECT 'churn_value', CONCAT(MIN(churn_value), ' - ' , MAX(churn_value))
  FROM customer_churn_status
 UNION ALL
SELECT 'churn_score', CONCAT(MIN(churn_score), ' - ' , MAX(churn_score))
  FROM customer_churn_status
 UNION ALL
SELECT 'cltv', CONCAT(MIN(cltv), ' - ' , MAX(cltv))
  FROM customer_churn_status
 UNION ALL
SELECT 'churn_category', STRING_AGG(DISTINCT churn_category, ', ')
  FROM customer_churn_status
 UNION ALL
SELECT 'churn_reason', STRING_AGG(DISTINCT churn_reason, ', ')
  FROM customer_churn_status;

 * postgresql://postgres:***@localhost:5432/telco
46 rows affected.


[('age_range', '19-80'),
 ('lat_long_range', '32.555828, -117.040073 - 41.962127, -122.527696'),
 ('zip_code_range', '90001 - 96150'),
 ('total_revenue', '21.36 - 11979.34'),
 ('total_long_distance_charges', '0 - 3564.72'),
 ('total_extra_data_charges', '0 - 150'),
 ('total_refund', '0 - 49.79'),
 ('total_charges', '18.8 - 8684.8'),
 ('monthly_charges', '18.25 - 118.75'),
 ('avg_monthly_gb_download', '0 - 85'),
 ('avg_monthly_long_distance', '0 - 49.99'),
 ('tenure_in_month', '1 - 72'),
 ('no_referred', '0 - 11'),
 ('cltv', '2003 - 6500'),
 ('churn_score', '5 - 96'),
 ('churn_value', '0 - 1'),
 ('satisfication_score', '1 - 5'),
 ('gender_range', 'Female, Male'),
 ('dependents_range', 'No, Yes'),
 ('married_range', 'No, Yes'),
 ('senior_citize_range', 'No, Yes'),
 ('under_30_range', 'No, Yes'),
 ('state', 'California'),
 ('country', 'United States'),
 ('payment_method', 'Bank Withdrawal, Credit Card, Mailed Check'),
 ('paperless_billing', 'No, Yes'),
 ('contract', 'Month-to-Month, One Year, Two Year'),
 ('customer_status', 'Churned, Joined, Stayed'),
 ('churn_category', 'Attitude, Competitor, Dissatisfaction, NA, Other, Price'),
 ('quarter', 'Q3'),
 ('offer', 'None, Offer A, Offer B, Offer C, Offer D, Offer E'),
 ('multiple_lines', 'No, Yes'),
 ('internet_type', 'Cable, DSL, Fiber Optic, None'),
 ('device_protection', 'No, Yes'),
 ('streaming_tv', 'No, Yes'),
 ('streaming_music', 'No, Yes'),
 ('churn_label', 'No, Yes'),
 ('churn_reason', "Attitude of service provider, Attitude of support person, Competitor had better devices, Competitor made better offer, Competitor offered higher down ... (224 characters truncated) ... work reliability, Poor expertise of online support, Poor expertise of phone support, Price too high, Product dissatisfaction, Service dissatisfaction"),
 ('referred_friend', 'No, Yes'),
 ('phone_service', 'No, Yes'),
 ('internet_service', 'No, Yes'),
 ('online_security', 'No, Yes'),
 ('online_backup', 'No, Yes'),
 ('premium_tech_support', 'No, Yes'),
 ('streaming_movies', 'No, Yes'),
 ('unlimited_data', 'No, Yes')]

While looking at the output, we can surely say the data is pretty clean, without any mistakes, which we need to take care of it.
I don't find any outliers in the qualitative data, but I find some thing odd in `total_long_distance_charges`,  `total_revenue` and `total_charges`.

Apart from that, everything is good to go.

So now we will check the data under `total_long_distance_charges`, `total_revenue` and `total_charges` to see if it makes sense.

In [ ]:
%%sql
SELECT total_long_distance_charges, total_revenue, total_charges
  FROM customer_churn_service
 WHERE total_long_distance_charges > (SELECT PERCENTILE_CONT(0.75) WITHIN GROUP(ORDER BY total_long_distance_charges) * 1.5
										                    FROM customer_churn_service)
 LIMIT 10;

 * postgresql://postgres:***@localhost:5432/telco
10 rows affected.


[(2355.1, 7319.8, 4874.7),
 (2732.8, 8527.45, 5794.65),
 (2055.9, 7665.2, 5509.3),
 (2864.04, 9151.29, 6287.25),
 (2844.87, 8259.33, 5427.05),
 (1904.76, 8726.91, 6822.15),
 (3372.72, 10690.92, 7318.2),
 (1936.48, 7447.13, 5510.65),
 (1980.0, 5620.45, 3640.45),
 (2259.06, 6753.62, 4541.2)]

As the long_distance_charges increase, ultimately it's having a positive effect on `total_revenue` and `total_charges`, which is concluding that the high range of values in `total_long_distance_charges`, `total_revenue`, and `total_charges` is fair.


### Data Exploration

After understanding what holds under the different tables, I'm ready to run different SQL queries that will try to bring insights from the underlying data and make stakeholders take appropriate decisions that will help to reduce the churn rate and increase revenue with a happy customer base.

### KPI

For the business, KPIs are a good starting point to see how things are moving, which helps them get the answer of whether they are on track to meet their goals or if adjustments need to be made.

To get a bigger picture of the datasets, I have taken the columns that speak about the churn, like the number of users who have churned, their satisfaction level with the brand, how much they were paying to the brand to get service, and importantly, how long customers have been associated with the business before it stopped their subscriptions.

In [ ]:
%%sql
/* Total Customer Churned */
SELECT 'total_customer_churned' AS KPI, COUNT(churn_label) AS value
  FROM customer_churn_status
 WHERE churn_label = 'Yes'
 UNION ALL
/* Churn Rate in */
SELECT 'churn_rate_in %', ROUND(COUNT(churn_label) * 100.0 / (SELECT COUNT(churn_label)
                                                                FROM customer_churn_status), 2)
  FROM customer_churn_status
 WHERE churn_label = 'Yes'
 UNION ALL
/* CSAT Score */
SELECT 'csat_score ☆ ', ROUND(AVG(satisfication_score),2)  AS csat_score
  FROM customer_churn_status
 WHERE churn_label = 'Yes'
 UNION ALL
/*Average Customer Tenure */
SELECT 'average_tenure_in_month 🗓', PERCENTILE_CONT(0.50) WITHIN GROUP (ORDER BY tenure_in_month)
  FROM customer_churn_service
 INNER JOIN customer_churn_status
    ON customer_churn_service.customer_id = customer_churn_status.customer_id
 WHERE churn_label = 'Yes'
 UNION ALL
/* Monthly Charges */
SELECT 'total_monthly_charges $', SUM(CAST(monthly_charges AS NUMERIC)) AS total_monthly_charges
  FROM customer_churn_service
 WHERE customer_id IN (SELECT customer_id
					               FROM customer_churn_status
					              WHERE churn_label = 'Yes')

 * postgresql://postgres:***@localhost:5432/telco
5 rows affected.


[('total_customer_churned', 1869.0),
 ('churn_rate_in %', 26.54),
 ('csat_score ☆ ', 1.74),
 ('average_tenure_in_month 🗓', 10.0),
 ('total_monthly_charges $', 139130.85)]

The results clearly show that customers are averagely ending their service within 10 months of the service, and the customer who churned also reported a satisfaction rating of just 1.7, which is not a good indicator for the business.

Also, it was reported from the 7043 customer count that it came to 5174 with a churn rate of 26.54%.

### Customer Tenure


Next, I'll figure out what the count of users who have already churned in the business is. From the above analysis, we get to know that the user just stayed for 10 months, and after that, they stopped using their service.
Is it just happening for the first year of the customer, or is the trend similar to other years too?



To clear the hypothesis let me tweak the into table by adding new column with the "tenure_in_year" for this I need to divide the tenure_in_month data by 12 and save the result into newly created columns.


In [ ]:
%%sql
ALTER TABLE customer_churn_service
ADD COLUMN tenure_in_year NUMERIC

 * postgresql://postgres:***@localhost:5432/telco
Done.


[]

In [ ]:
%%sql
UPDATE customer_churn_service
SET tenure_in_year = ROUND((tenure_in_month / 12.0), 2);

 * postgresql://postgres:***@localhost:5432/telco
7043 rows affected.


[]

With the above query I have sucessfully added the new column to customer_churn_service. Now using the tenure_in_year column will put the customer into different bin according to time span with business, and then will check the count, and churn percentage.

In [ ]:
%%sql
SELECT
	 CASE
	 WHEN tenure_in_year <= 1 THEN 'Between 0 and 1 Year'
	 WHEN tenure_in_year < 1 AND tenure_in_year <= 2 THEN 'Between 1 and 2 Years'
	 WHEN tenure_in_year < 2 AND tenure_in_year <= 3 THEN 'Between 2 and 3 Years'
	 WHEN tenure_in_year < 3 AND tenure_in_year <= 4 THEN 'Between 3 and 4 Years'
	 WHEN tenure_in_year < 4 AND tenure_in_year <= 5 THEN 'Between 4 and 5 Years'
	 ELSE 'Between 6 and More Years'
	 END AS tenure_period,
	 COUNT(*) AS frequency,
	 CONCAT(ROUND(COUNT(*)  * 100.0 / (SELECT COUNT(*)
                                       FROM customer_churn_status
								                      WHERE churn_label = 'Yes'),2), '%') AS percentage
  FROM customer_churn_service
 INNER JOIN customer_churn_status
    ON customer_churn_service.customer_id = customer_churn_status.customer_id
 WHERE churn_label = 'Yes'
 GROUP BY tenure_period
 ORDER BY tenure_period ASC;

 * postgresql://postgres:***@localhost:5432/telco
5 rows affected.


[('Between 0 and 1 Year', 1037, '55.48%'),
 ('Between 2 and 3 Years', 271, '14.50%'),
 ('Between 3 and 4 Years', 193, '10.33%'),
 ('Between 4 and 5 Years', 146, '7.81%'),
 ('Between 6 and More Years', 222, '11.88%')]

It's also showing that people who are engaged in the business within a year are more prone to churn than any other group. <br>
As the tenure gets more mature, the count of churn is showing a downward trend, but I see a spike in the churn rate again when customers are between their 6 and more years of business.

### Demographic Information

Next, we will see which age group of people tends to churn compared to others, and then it will help us target a new market segment to a specific age of users to get the maximum result from them.

In [ ]:
%%sql
SELECT
       CASE
          WHEN age BETWEEN 00 AND 10 THEN '0-10'
          WHEN age BETWEEN 11 AND 20 THEN '11-20'
          WHEN age BETWEEN 21 AND 30 THEN '21-30'
          WHEN age BETWEEN 31 AND 40 THEN '31-40'
          WHEN age BETWEEN 41 AND 50 THEN '41-50'
          WHEN age BETWEEN 51 AND 60 THEN '51-60'
          WHEN age BETWEEN 61 AND 70 THEN '61-70'
          WHEN age BETWEEN 71 AND 80 THEN '71-80'
          WHEN age BETWEEN 81 AND 90 THEN '81-90'
          ELSE 'Super Senior Citizen'
        END age_bin,
        COUNT(*),
        CONCAT(ROUND(COUNT(*)  * 100.0 / (SELECT COUNT(*)
                                            FROM customer_churn_status
								                           WHERE churn_label = 'Yes'),2), '%') AS percentage
  FROM customer_churn_demographic
 INNER JOIN customer_churn_status
    ON customer_churn_demographic.customer_id = customer_churn_status.customer_id
 WHERE churn_label = 'Yes'
 GROUP BY age_bin
 ORDER BY age_bin

 * postgresql://postgres:***@localhost:5432/telco
7 rows affected.


[('11-20', 48, '2.57%'),
 ('21-30', 294, '15.73%'),
 ('31-40', 303, '16.21%'),
 ('41-50', 309, '16.53%'),
 ('51-60', 309, '16.53%'),
 ('61-70', 313, '16.75%'),
 ('71-80', 293, '15.68%')]

There is no significant change in churn with respect to age over churn, which means no matter what age group the customer is can churn from business anytime.

### Gender Distributions

As of now, we know there were a total of 7043 customers, and subsequently, the count came to 5041, but we are not yet aware which gender has churned more: male or female? and what is their total churn rate?
To unveil this information, let me run this SQL query to show the churn by gender distributions.

In [ ]:
%%sql
SELECT gender,
	     total AS initial_customer,
	     total - churned_count AS cutomer_left_after_churn,
		   churned_count AS total_churned_customer,
	     CONCAT(ROUND((churned_count * 100.0 / (SELECT COUNT(customer_id)
												 FROM customer_churn_demographic)),2),'%') AS churn_percentage
  FROM (SELECT gender,
		          COUNT(*) AS total,
	     (SELECT COUNT(*)
	        FROM customer_churn_status
	       WHERE churn_label = 'Yes' AND customer_id IN (SELECT customer_id
												  	                             FROM customer_churn_demographic
												  	                            WHERE gender = c.gender)
	    )AS churned_count
	   FROM customer_churn_demographic c
	  GROUP BY gender
);

 * postgresql://postgres:***@localhost:5432/telco
2 rows affected.


[('Female', 3488, 2549, 939, '13.33%'), ('Male', 3555, 2625, 930, '13.20%')]

This correlated subquery is showing us the initital_customer count, and after that, it shows the number of customers left after churn by the gender columns, which is showing both genders are showing the same behavior towards the brand.<br>
Is it the same thing we will see if we analyze the senior citizen group?

In [ ]:
%%sql
SELECT gender,
		COUNT(senior_citizen) AS total_churned,
		CONCAT(ROUND(COUNT(senior_citizen) * 100.0 /
			  (SELECT COUNT(*)
 				   FROM customer_churn_status
				  WHERE churn_label = 'Yes'), 2), '%') AS senior_citizen_churned
 FROM customer_churn_demographic
WHERE senior_citizen = 'Yes' AND customer_id IN (SELECT customer_id
												                           FROM customer_churn_status
												                          WHERE churn_label = 'Yes')
GROUP BY gender

 * postgresql://postgres:***@localhost:5432/telco
2 rows affected.


[('Female', 240, '12.84%'), ('Male', 236, '12.63%')]

Here also, we see the same trend: equally, both gender groups have churned.

The data has considered senior citizens who are more than 65 years old, and as you know, as the age increases, the life span starts to shorten, and it may be due to that senior citizens customers may be churned, but as per the [CDC life expectancy](https://www.cdc.gov/nchs/pressroom/sosmap/life_expectancy/life_expectancy.htm) in California is 79.0 years, and we have the person till age 80, so let's run the query and find what the truth is.

In [ ]:
%%sql
SELECT COUNT(senior_citizen),
		CONCAT(ROUND(COUNT(senior_citizen) * 100.0 /
			  (SELECT COUNT(*)
 				 FROM customer_churn_status
				WHERE churn_label = 'Yes'), 2), '%') AS senior_citizen_churned
  FROM customer_churn_demographic
 WHERE senior_citizen = 'Yes' AND customer_id IN (SELECT customer_id
												 													  FROM customer_churn_status
												                            WHERE churn_label = 'Yes' AND churn_reason = 'Deceased')

 * postgresql://postgres:***@localhost:5432/telco
1 rows affected.


[(2, '0.11%')]

Only two people have closed the service due to a decased reason; from 473 people, now the rest may have dropped the connection due to some other reason. <br>
If the deceased is not the main cause of the churned-in senior citizen count, then what is it?

In [ ]:
%%sql
SELECT churn_reason,
		COUNT(senior_citizen) AS frequency,
		CONCAT(ROUND(COUNT(senior_citizen) * 100.0 /
			  (SELECT COUNT(*)
 				 FROM customer_churn_status
				WHERE churn_label = 'Yes'), 2), '%') AS senior_citizen_churned
  FROM customer_churn_demographic
 INNER JOIN customer_churn_status
    ON customer_churn_demographic.customer_id = customer_churn_status.customer_id
 WHERE senior_citizen = 'Yes'
   AND customer_churn_demographic.customer_id IN (SELECT customer_id
												                            FROM customer_churn_status
												                           WHERE churn_label = 'Yes')
 GROUP BY churn_reason
 ORDER BY frequency DESC

 * postgresql://postgres:***@localhost:5432/telco
19 rows affected.


[('Competitor made better offer', 90, '4.82%'),
 ('Competitor had better devices', 84, '4.49%'),
 ('Attitude of support person', 64, '3.42%'),
 ('Competitor offered more data', 27, '1.44%'),
 ('Price too high', 26, '1.39%'),
 ("Don't know", 26, '1.39%'),
 ('Attitude of service provider', 25, '1.34%'),
 ('Product dissatisfaction', 23, '1.23%'),
 ('Network reliability', 18, '0.96%'),
 ('Limited range of services', 15, '0.80%'),
 ('Service dissatisfaction', 15, '0.80%'),
 ('Competitor offered higher download speeds', 12, '0.64%'),
 ('Extra data charges', 12, '0.64%'),
 ('Long distance charges', 10, '0.54%'),
 ('Poor expertise of online support', 10, '0.54%'),
 ('Moved', 8, '0.43%'),
 ('Lack of affordable download/upload speed', 8, '0.43%'),
 ('Deceased', 2, '0.11%'),
 ('Poor expertise of phone support', 1, '0.05%')]

The most common reason for switching providers is that competitors made a better offer to them with better devices, more data, and so on, which clearly speaks to the churn reasons of most churn is competitior. We should not ignore the attitude of the support person toward the customer, which is also contributing enough to increase the churn count.

The same thing is happening with the other customer base.

In [ ]:
%%sql
SELECT churn_reason,
		   COUNT(gender) AS frequency,
		   CONCAT(ROUND(COUNT(gender) * 100.0 /
			  (SELECT COUNT(*)
 				   FROM customer_churn_status
				  WHERE churn_label = 'Yes'), 2), '%') AS total_churned
  FROM customer_churn_demographic
 INNER JOIN customer_churn_status
    ON customer_churn_demographic.customer_id = customer_churn_status.customer_id
 WHERE churn_label = 'Yes'
 GROUP BY churn_reason
 ORDER BY frequency DESC
 LIMIT 5

 * postgresql://postgres:***@localhost:5432/telco
5 rows affected.


[('Competitor had better devices', 313, '16.75%'),
 ('Competitor made better offer', 311, '16.64%'),
 ('Attitude of support person', 220, '11.77%'),
 ("Don't know", 130, '6.96%'),
 ('Competitor offered more data', 117, '6.26%')]

With these top 5 reasons, we again concluded that competitors are the main area where brands need to work, and the same thing repeated when the customer is a senior citizen.

In [ ]:
%%sql
SELECT COALESCE(churn_reason, 'Total') AS reasons,
	   COUNT(senior_citizen) AS frequency,
	   CONCAT(ROUND(COUNT(senior_citizen) * 100.0 /
			  (SELECT COUNT(*)
 				   FROM customer_churn_status
				  WHERE churn_label = 'Yes'), 2), '%') AS senior_citizen_churned
  FROM customer_churn_demographic
 INNER JOIN customer_churn_status
    ON customer_churn_demographic.customer_id = customer_churn_status.customer_id
 WHERE senior_citizen = 'Yes'
   AND customer_churn_demographic.customer_id IN (SELECT customer_id
												                            FROM customer_churn_status
												                           WHERE churn_label = 'Yes' AND churn_reason LIKE 'Competitor%' )
GROUP BY CUBE (churn_reason)
ORDER BY frequency ASC

 * postgresql://postgres:***@localhost:5432/telco
5 rows affected.


[('Competitor offered higher download speeds', 12, '0.64%'),
 ('Competitor offered more data', 27, '1.44%'),
 ('Competitor had better devices', 84, '4.49%'),
 ('Competitor made better offer', 90, '4.82%'),
 ('Total', 213, '11.40%')]

Is the same thing true when the CLTV is high and the average rating of satisfaction is for other age groups?

In [ ]:
%%sql
SELECT COALESCE(churn_reason, 'Total') AS reasons,
		   COUNT(gender) AS frequency,
	     CONCAT(ROUND(COUNT(*) * 100.0 /
			  (SELECT COUNT(*)
 				  FROM customer_churn_status
				 WHERE churn_label = 'Yes'), 2), '%') AS user_churned
  FROM customer_churn_demographic
 INNER JOIN customer_churn_status
    ON customer_churn_demographic.customer_id = customer_churn_status.customer_id
 WHERE churn_label = 'Yes' AND satisfication_score >= 4
   AND customer_churn_demographic.customer_id IN (SELECT customer_id
												                            FROM customer_churn_status
                                                   GROUP BY customer_id, cltv
                                                  HAVING cltv >= PERCENTILE_CONT(0.75) WITHIN GROUP(ORDER BY cltv))
 GROUP BY gender, churn_reason
 ORDER BY frequency DESC
 LIMIT 5

 * postgresql://postgres:***@localhost:5432/telco
0 rows affected.


[]

There is no result when the satisfaction level is 4 or greater than 4 or when the CLTV is high.<br>
This clearly indicates that when the customer is happy with the service, the chance of churn is very low.

Let me keep the query the same and just change the `satisfaction_score` to 3, and see what it shows.

In [ ]:
%%sql
SELECT COALESCE(churn_reason, 'Total') AS reasons,
		 COUNT(gender) AS frequency,
	    CONCAT(ROUND(COUNT(*) * 100.0 /
			  (SELECT COUNT(*)
 				  FROM customer_churn_status
				 WHERE churn_label = 'Yes'), 2), '%') AS user_churned
  FROM customer_churn_demographic
 INNER JOIN customer_churn_status
    ON customer_churn_demographic.customer_id = customer_churn_status.customer_id
 WHERE churn_label = 'Yes' AND satisfication_score >= 3
   AND customer_churn_demographic.customer_id IN (SELECT customer_id
												                            FROM customer_churn_status
                                                   GROUP BY customer_id, cltv
                                                  HAVING cltv >= PERCENTILE_CONT(0.75) WITHIN GROUP(ORDER BY cltv))
 GROUP BY churn_reason
 ORDER BY frequency DESC
 LIMIT 5

 * postgresql://postgres:***@localhost:5432/telco
5 rows affected.


[('Competitor made better offer', 87, '4.65%'),
 ('Attitude of support person', 70, '3.75%'),
 ('Competitor offered more data', 36, '1.93%'),
 ('Attitude of service provider', 35, '1.87%'),
 ("Don't know", 27, '1.44%')]

As we lowered the satisfaction level and kept the CLTV level at 75%, it shows the top 5 reasons where it's equally pointing out that the competitor and attitude of the support person are causing real harm to business.<br>
To get the picture more clearly, we set the satisfaction score to 1 and kept the CLTV high to see what the reason was for the high CLTV customer to drop the service.

In [ ]:
%%sql
SELECT COALESCE(churn_reason, 'Total') AS reasons,
		 COUNT(gender) AS frequency,
	    CONCAT(ROUND(COUNT(*) * 100.0 /
			  (SELECT COUNT(*)
 				  FROM customer_churn_status
				 WHERE churn_label = 'Yes'), 2), '%') AS user_churned
  FROM customer_churn_demographic
 INNER JOIN customer_churn_status
    ON customer_churn_demographic.customer_id = customer_churn_status.customer_id
 WHERE churn_label = 'Yes' AND satisfication_score >= 1
   AND customer_churn_demographic.customer_id IN (SELECT customer_id
												                            FROM customer_churn_status
                                                   GROUP BY customer_id, cltv
                                                  HAVING cltv >= PERCENTILE_CONT(0.75) WITHIN GROUP(ORDER BY cltv))
 GROUP BY churn_reason
 ORDER BY frequency DESC
 LIMIT 5

 * postgresql://postgres:***@localhost:5432/telco
5 rows affected.


[('Competitor had better devices', 313, '16.75%'),
 ('Competitor made better offer', 311, '16.64%'),
 ('Attitude of support person', 220, '11.77%'),
 ("Don't know", 130, '6.96%'),
 ('Competitor offered more data', 117, '6.26%')]

From the above result, we find competitors are one of the biggest challenges because that customer has dropped the connection.

In [ ]:
%%sql
SELECT  COALESCE(churn_reason, 'Total') AS reasons,
		COUNT(*) AS frequency,
	    CONCAT(ROUND(COUNT(*) * 100.0 /
			  (SELECT COUNT(*)
 				  FROM customer_churn_status
				 WHERE churn_label = 'Yes'), 2), '%') AS user_churned
  FROM customer_churn_demographic
 INNER JOIN customer_churn_status
    ON customer_churn_demographic.customer_id = customer_churn_status.customer_id
   AND customer_churn_demographic.customer_id IN (SELECT customer_id
												    FROM customer_churn_status
												   WHERE churn_label = 'Yes' AND churn_reason LIKE 'Competitor%' )
 GROUP BY CUBE (churn_reason)
 ORDER BY frequency ASC

 * postgresql://postgres:***@localhost:5432/telco
5 rows affected.


[('Competitor offered higher download speeds', 100, '5.35%'),
 ('Competitor offered more data', 117, '6.26%'),
 ('Competitor made better offer', 311, '16.64%'),
 ('Competitor had better devices', 313, '16.75%'),
 ('Total', 841, '45.00%')]

Now it's clear that the competitor is the main root cause of churn among high CLTV customers. <br>
As you know, brands provide three types of contracts: month-to-month, one-year, and two-year. From this information, let's see which category has the highest churn rate.

In [ ]:
%%sql
SELECT contract,
	   COUNT(contract) AS contract_count,
	   CONCAT(COUNT(contract) * 100 / (SELECT COUNT(*)
								  FROM customer_churn_status
							     WHERE churn_label = 'Yes' ), '%') AS churn_in_percentage
  FROM customer_churn_service
 WHERE customer_id IN (SELECT customer_id
						 FROM customer_churn_status
						WHERE churn_label = 'Yes')
 GROUP BY contract
 ORDER BY churn_in_percentage DESC

 * postgresql://postgres:***@localhost:5432/telco
3 rows affected.


[('Month-to-Month', 1655, '88%'),
 ('One Year', 166, '8%'),
 ('Two Year', 48, '2%')]

This result shows that the month-to-month contract is the most preferable plan for the consumer who gets churned.

In [ ]:
%%sql
 WITH top_3_reason AS (
    SELECT contract,
           churn_reason,
           ROW_NUMBER() OVER(PARTITION BY contract ORDER BY COUNT(contract) DESC) as position,
           COUNT(contract) AS contract_count,
           COUNT(contract) * 100.0 / (SELECT COUNT(*)
                                        FROM customer_churn_status
                                       WHERE churn_label = 'Yes') AS churn_in_percentage
      FROM customer_churn_service
     INNER JOIN customer_churn_status
        ON customer_churn_service.customer_id = customer_churn_status.customer_id
     WHERE customer_churn_status.churn_label = 'Yes'
     GROUP BY contract, churn_reason
)
SELECT
    contract,
    churn_reason,
    contract_count,
    ROUND(churn_in_percentage, 2) AS churn_percentage
  FROM top_3_reason
 WHERE position <=3
 ORDER BY contract, position;

 * postgresql://postgres:***@localhost:5432/telco
9 rows affected.


[('Month-to-Month', 'Competitor had better devices', 288, Decimal('15.41')),
 ('Month-to-Month', 'Competitor made better offer', 280, Decimal('14.98')),
 ('Month-to-Month', 'Attitude of support person', 205, Decimal('10.97')),
 ('One Year', 'Competitor made better offer', 23, Decimal('1.23')),
 ('One Year', 'Competitor had better devices', 19, Decimal('1.02')),
 ('One Year', 'Competitor offered higher download speeds', 15, Decimal('0.80')),
 ('Two Year', 'Competitor made better offer', 8, Decimal('0.43')),
 ('Two Year', 'Competitor had better devices', 6, Decimal('0.32')),
 ('Two Year', 'Competitor offered more data', 6, Decimal('0.32'))]

The top reason in every contract is a competitor, and it's very likely that customers who are in a month-to-month contract will switch the connection if they get better devices with a better offer from competitors.

Now let's add new twist to the analysis by checking the ratio of churn between customers who are married and single.

In [ ]:
%%sql
 SELECT married,
        COUNT(married),CONCAT(ROUND(COUNT(gender) * 100.0 /
		  (SELECT COUNT(*)
 			  FROM customer_churn_status
          WHERE churn_label = 'Yes'), 2), '%') AS total_churned
   FROM customer_churn_demographic
  INNER JOIN customer_churn_status
     ON customer_churn_demographic.customer_id = customer_churn_status.customer_id
  WHERE churn_label = 'Yes'
  GROUP BY married

 * postgresql://postgres:***@localhost:5432/telco
2 rows affected.


[('No', 1200, '64.21%'), ('Yes', 669, '35.79%')]

The above query output shows unmarried people are more prone to churn than comparatively married people, which takes us to the question next: What would be the dependency if the person is married or unmarried? As you know, after marriage, some kind of dependency is added to an individual.

In [ ]:
%%sql
SELECT married, dependents, COUNT(married) AS frequency
  FROM customer_churn_demographic
 INNER JOIN customer_churn_status
    ON customer_churn_demographic.customer_id = customer_churn_status.customer_id
 WHERE churn_label = 'Yes'
 GROUP BY married, dependents

 * postgresql://postgres:***@localhost:5432/telco
4 rows affected.


[('No', 'Yes', 50), ('Yes', 'Yes', 56), ('No', 'No', 1150), ('Yes', 'No', 613)]

It is clearly showing that after marriage, it's not always true to get dependent on the partner.<br>
Also, it highlights that when the customer is not married and dependents, they are highly likely to switch connections.

Up to this point, we have examined various factors, such as the primary reasons for customer churn, the average duration customers stay with businesses, and other related aspects. However, we have not yet analyzed which of the top 10 cities are influencing churn rates. To address this, let's execute the next query.

In [ ]:
%%sql
  SELECT city, COUNT(city) AS city_frequency
  	 FROM customer_churn_location
   INNER JOIN customer_churn_status
      ON customer_churn_location.customer_id = customer_churn_status.customer_id
   WHERE churn_label = 'Yes'
   GROUP BY city
   ORDER BY city_frequency DESC
   LIMIT 10

 * postgresql://postgres:***@localhost:5432/telco
10 rows affected.


[('San Diego', 185),
 ('Los Angeles', 78),
 ('San Francisco', 31),
 ('San Jose', 29),
 ('Fallbrook', 26),
 ('Sacramento', 26),
 ('Temecula', 22),
 ('Escondido', 16),
 ('Long Beach', 15),
 ('Fresno', 13)]

In the top 10 cities, we get San Diego is on top with 185 more people churned than any other city, following Los Angeles, San Francisco, and so on.

Now, we will retain the same cities from the previous analysis and determine which city is most influenced by competitors.

In [ ]:
%%sql
   SELECT city, COUNT(city) AS city_frequency
  	  FROM customer_churn_location
    INNER JOIN customer_churn_status
       ON customer_churn_location.customer_id = customer_churn_status.customer_id
    WHERE churn_label = 'Yes' AND churn_reason LIKE 'C%' AND city IN ('San Diego', 'Los Angeles', 'San Francisco', 'San Jose', 'Fallbrook', 'Sacramento', 'Temecula', 'Escondido', 'Long Beach', 'Fresno')
    GROUP BY city
    ORDER BY city_frequency DESC
    LIMIT 10

 * postgresql://postgres:***@localhost:5432/telco
10 rows affected.


[('San Diego', 152),
 ('Los Angeles', 39),
 ('Fallbrook', 23),
 ('Temecula', 18),
 ('Sacramento', 18),
 ('San Francisco', 17),
 ('San Jose', 15),
 ('Escondido', 14),
 ('Long Beach', 3),
 ('Fresno', 2)]

Here also, San Diego has a high churned count because of competitors.

We have explored various aspects, but we haven't explored which service has the highest effect of churn on the service reduction that Fintel offers to its customers.

In [ ]:
%%sql
SELECT
    COUNT(phone_service) * 100 / (SELECT count(*)
                                    FROM customer_churn_service
                                   WHERE phone_service = 'Yes') AS "phone_service_churned_in_%",
    COUNT(multiple_lines) * 100 / (SELECT count(*)
                                     FROM customer_churn_service
                                    WHERE multiple_lines = 'Yes') AS "multiple_line_churned_in_%",
    COUNT(internet_service) * 100 / (SELECT count(*)
                                       FROM customer_churn_service
                                      WHERE internet_service = 'Yes') AS "internet_service_churned_in_%",
    COUNT(online_security) * 100 / (SELECT count(*)
                                      FROM customer_churn_service
                                     WHERE online_security = 'Yes') AS "online_security_churned_in_%",
    COUNT(online_backup) * 100 / (SELECT count(*)
                                    FROM customer_churn_service
                                   WHERE online_backup = 'Yes') AS "online_backup_churned_in_%",
    COUNT(device_protection) * 100 / (SELECT count(*)
                                        FROM customer_churn_service
                                       WHERE device_protection = 'Yes') AS "device_protection_in_%",
    COUNT(premium_tech_support) * 100 / (SELECT count(*)
                                           FROM customer_churn_service
                                          WHERE premium_tech_support = 'Yes') AS "premium_tech_support_churned_in_%",
    COUNT(streaming_tv) * 100 / (SELECT count(*)
                                   FROM customer_churn_service
                                  WHERE streaming_tv = 'Yes') AS "streaming_tv_churned_in_%",
    COUNT(streaming_movies) * 100 / (SELECT count(*)
                                       FROM customer_churn_service
                                      WHERE streaming_movies = 'Yes') AS "streaming_movies_churned_in_%",
    COUNT(streaming_music) * 100 / (SELECT count(*)
                                      FROM customer_churn_service
                                     WHERE streaming_music = 'Yes') AS "streaming_music_churned_in_%"
  FROM customer_churn_service
 INNER JOIN customer_churn_status
    ON customer_churn_service.customer_id = customer_churn_status.customer_id
 WHERE churn_label = 'Yes'

 * postgresql://postgres:***@localhost:5432/telco
1 rows affected.


[(29, 62, 33, 92, 76, 77, 91, 69, 68, 75)]

The results clearly indicate that the majority of customers who have churned have discontinued their phone and internet services.

As the customer churns, it impacts the revenue, and here we have the average total charges and the monthly charges that the customer pays, and from this, let's  take out the difference between the two to determine the potential loss in revenue due to customer churn.

Average total & monthly charges with the churn effect

In [ ]:
%%sql
SELECT PERCENTILE_CONT(0.50) WITHIN GROUP (ORDER BY total_charges) AS avg_total_charges,
       PERCENTILE_CONT(0.50) WITHIN GROUP (ORDER BY monthly_charges) AS  avg_monthly_charges
  FROM customer_churn_service
 INNER JOIN customer_churn_status
    ON customer_churn_service.customer_id = customer_churn_status.customer_id
 WHERE churn_label = 'Yes'

 * postgresql://postgres:***@localhost:5432/telco
1 rows affected.


[(703.55, 79.65)]

Average total & monthly charges without the churn effect

In [ ]:
%%sql
SELECT PERCENTILE_CONT(0.50) WITHIN GROUP (ORDER BY total_charges) AS avg_total_charges,
       PERCENTILE_CONT(0.50) WITHIN GROUP (ORDER BY monthly_charges) AS  avg_monthly_charges
  FROM customer_churn_service
 INNER JOIN customer_churn_status
    ON customer_churn_service.customer_id = customer_churn_status.customer_id
 WHERE churn_label = 'No'

 * postgresql://postgres:***@localhost:5432/telco
1 rows affected.


[(1679.525, 64.42500000000001)]

From both sets of results, it's apparent that the average monthly charges are high due to the churned users mainly subscribing to monthly plans.
On the other hand, customers who didn't churn generally exhibit lower average monthly charges. However, there's a notable contrast in total charges between these two groups.

## Insights

I know everyone is waiting eagerly for the insights we have received from the datasets that can be used by our fictional company (Fictel) to see how things are moving and what they can do to do their best to attract and prevent a lower churn rate, which ultimately increases profit, and positive word-of-mouth marketing.

While working with the dataset size of 7043 unique customers with 63 columns that have different attributes about each customer

To summarize all this, we'll focus on the most crucial KPIs, offering a clear picture of churn within Fictel and demonstrating a strong understanding of the situation.

The first and foremost thing is to notice that around 1869 customers have churned from the total count of 7043, which comes to a churn rate of 26.54% with an average CSAT score of 1.74.
Also,  the customer stays with Fictel just for 10 months, and this churned customer has added $ 139,130 to revenue.

However 55% of Fictel users are not moving with the brand for the next year, and they discontinue service due to several reasons, the most prominent of which is due to competitors and the behavior of staff.

As the median age of customers is around 45, the average age of customers who are churned is 50, and the churn ratio is the same regardless of gender.

Churned customers typically opt for month-to-month contracts, and their reasons for churn often align with competitors offering better devices, deals, and occasionally inadequate support during their tenure as customers.

Intrestingly, data also speaks the same: married individuals are less likely to switch networks as compared to non-married people, so the company should make sure the married customers are happy with the service.
As you will find, the churn rate is 35.79%, compared to 64.21% in non-married individuals.

From the 500+ cities in California, Fictel data indicates that the people of San Diego have churned from the connection of 285; now it's just left 185, and the similar data point for Los Angeles shows that only 78 customers have churned, which means San Diego customers are not happy with the service.

Also, we can see the churned percentage in terms of service as follows:

| Service | Churned % |
| :- | -: |
| Phone Service | 29 |
| Multiple Line Service| 32 |
| Internet Service| 33 |
| Online Security Service| 92 |
| Online Backup Service| 76 |
| Device Protection Service| 77 |
| Premium Tech Support Service | 91 |
| Streaming TV Service | 69 |
| Streaming Movies Service| 68 |
| Streaming Music Service| 75 |

Additionally, we can see a significant difference in charges, like the customer who churned is paying average total charges of approximately \$703 and the customer who is not churned is valued at around $1680, and their average monthly charges come to \$79.65 and \$64 when the customer stays with the brand.

This is one of the few insights I have gained from the datasets that brands can use to understand where their business is headed and make necessary improvements to retain customers.

### Suggestions

There are a few suggestions I would like to bring here so that it would be helpful for the brand to make the right decision and make a profitable business with a happy customer base.

1. The first thing I suggest to you is to bring down the churn rate by giving the best service by giving them a better offer, data value, and so on, because the market is competitive and has to compete with the new technology.

2. Also train the staff to be more polite towards customers, and incentivize the employee with a good rating so they will be more happy to assist customers.

3. While analyzing, I found that the users who subscribe year-to-year are less likely to churn, so try to pitch them for a longer period.

4. On average, the churn user age is between 45-50, and with unmarried people churning, any medium, if we can monitor their satisfaction level, can make the churn rate even lower.

5. Even the marketing team can market their service to the 40–50 age group of people.

6. From the other big cities in California, San Diego, customers are getting more churned. If we have something special for this region, then we can have happy customers.

This was one of my few suggestions for Fictel.

Thank you for taking the time to read this analysis. If you want me to bring out more insights from the data that I missed here, then contact me on [LinkedIn](https://www.linkedin.com/in/gagandev-mishra/) or via [email](mailto:gagandev.mishra@gmail.com) so we can discuss further.